src/Http/Authentication.Core/src/AuthenticationSchemeProvider.cs

```diff
@@ -152,6 +153,26 @@ public virtual void AddScheme(AuthenticationScheme scheme)
                 }
                 _schemes[scheme.Name] = scheme;
                 _schemesCopy = _schemes.Values.ToArray();
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Registers a scheme for use by <see cref="IAuthenticationService"/>. 
+        /// </summary>
+        /// <param name="scheme">The scheme.</param>
+        public virtual void AddScheme(AuthenticationScheme scheme)
+        {
+            if (_schemes.ContainsKey(scheme.Name))
```

Comment:
This check is redundant, AddScheme will succeed most of the time so there's no point optimizing for the lock.